tests/ResultQueryTest.php

```diff
@@ -588,4 +588,164 @@ public function testQueryStreamFromLazyConnectionWillErrorWhenConnectionIsClosed
 
         $connection->close();
     }
+
+    protected function checkMaxAllowedPacket($connection, $min = 0x1100000)
+    {
+        return $connection->query('SHOW VARIABLES LIKE \'max_allowed_packet\'')->then(
+            function ($res) use ($min, $connection) {
+                $current = $res->resultRows[0]['Value'];
+                if ($current < $min) {
+                    throw new \Exception('max_allowed_packet too low: current: ' . $current . ' min: ' . $min);
+                }
+                return \React\Promise\resolve();
+            }
+        )->then(
+            function () {
+                return true;
+            }
+        );
+    }
+
+    /**
+     * This should not trigger splitted packets sending
+     */
+    public function testSelectStaticTextSplitPacketsExactlyBelow16MiB()
+    {
+        $connection = $this->createConnection(Loop::get());
+
+        $promise = $this->checkMaxAllowedPacket($connection, 0x1000000); // 16MiB
+
+        $promise->then(
+            function () use ($connection) {
+                /**
+                 * This should be exactly below 16MiB packet
+                 *
+                 * x03 + "select ''" = len(10)
+                 */
+                $text = str_repeat('A', 0xffffff - 11);
+                $connection->query('select \'' . $text . '\'')->then(function (QueryResult $command) use ($text) {
+                    $this->assertCount(1, $command->resultRows);
+                    $this->assertCount(1, $command->resultRows[0]);
+                    $this->assertSame($text, reset($command->resultRows[0]));
+                })->done();
+            }
+        )->otherwise(
+            function (\Throwable $e) {
+                $this->markTestIncomplete('checkMaxAllowedPacket: ' . $e->getMessage());
+            }
```

Comment:
I think we should skip these tests here instead of marking them as incomplete. As the [documentation of PHPUnit](https://phpunit.readthedocs.io/en/9.5/incomplete-and-skipped-tests.html#incomplete-tests) says to mark tests as incomplete:

> `PHPUnit\Framework\IncompleteTest` is a marker interface for marking an exception that is raised by a test method as the result of the test being incomplete or currently not implemented. 

The problem here isn't that some implementation is missing, it's more of a "Your configuration doesn't fit our test" case here. This fits the example for [skipping tests](https://phpunit.readthedocs.io/en/9.5/incomplete-and-skipped-tests.html#skipping-tests) with PHPUnit as described inside the docs:

>  In the test case class’ setUp() template method we check whether the MySQLi extension is available and use the markTestSkipped() method to skip the test if it is not.

Makes more sense, what do you think?